Tests/Common/IntegrationTestFixtureBase.cs

```diff
@@ -327,6 +327,25 @@ public async Task UpdateExistingConcentratorCrcValues(StationEui stationEui, uin
             await registryManager.UpdateTwinAsync(stationDeviceId, deviceTwin, deviceTwin.ETag);
         }
 
+        public async Task UpdateExistingFirmwareUpgradeValues(StationEui stationEui, uint crc, string digestBase64String)
+        {
+            TestLogger.Log($"Updating IoT Hub twin for fw upgrades of concentrator {stationEui}...");
+            var registryManager = GetRegistryManager();
+            var stationDeviceId = GetDeviceId(stationEui);
+            var getDeviceResult = await registryManager.GetDeviceAsync(stationDeviceId);
+            if (getDeviceResult == null)
+                throw new InvalidOperationException("Concentrator should exist in IoT Hub");
+            var deviceTwin = await registryManager.GetTwinAsync(stationDeviceId);
+            var cupsJson = ((object)deviceTwin.Properties.Desired[BasicsStationConfigurationService.CupsPropertyName]).ToString();
+            var newCupsInfo = JsonConvert.DeserializeObject<CupsTwinInfo>(cupsJson) with
```

Comment:
Shouldn't we also save new package version and firwmware location in storage?

Reply:
Done